nb-repository-plugin/src/main/java/org/apache/netbeans/nbm/repository/PopulateRepositoryMojo.java

```diff
@@ -1105,17 +1104,24 @@ else if ( wr instanceof ModuleWrapperMaven )
 //        mavenModel.setBuild(build);
 
         File fil = null;
-        try ( FileWriter writer = new FileWriter( fil ) )
+        try 
         {
-            MavenXpp3Writer xpp = new MavenXpp3Writer();
-            fil = Files.createTempFile( "maven", ".pom" ).toFile();
+            fil = Files.createTempFile("maven", ".pom").toFile();
             fil.deleteOnExit();
-            xpp.write( writer, mavenModel );
-        }
-        catch ( IOException ex )
+        } catch (IOException ex) 
         {
             ex.printStackTrace();
-        }        
+        }
+        if (fil != null) {
```

Comment:
Would this not be better as nested `try`?

```
try {
  File file = Files.createTempFile("maven", ".pom").toFile();
  file.deleteOnExit();
  try (FileWriter writer = new FileWriter(file)) {
     // write
  }
} catch (IOException ex) {
  // handle all exceptions
}
```

It's a shame try-with-resources doesn't seem to handle intermediate declarations that are not `AutoCloseable`.

Reply:
for me better but nested are often discouraged

Reply:
I think the arguments against nesting are lost the moment you have to work around it by doing `if (file != null)` :smile:

Reply:
i have no problems at all with nesting, since it is very easy to get out of the situation once the method nests too deep: extract pieces to new methods.

so nesting itself: no problem at all, if there is some periodic refactoring happening to keep methods simple.

the try-with-resource block is essentially a scope block. It will find more usecases with structured concurrency etc so there will be a lot more of them soon.